perceval/components/abstract_processor.py

```diff
@@ -443,271 +254,135 @@ def add_herald(self, mode: int, expected: int, name: str = None) -> AProcessor:
-            if mode in mode_range:
-                return port
-        return None
+        return self.experiment.get_input_port(mode)
 
     def get_output_port(self, mode):
-        for port, mode_range in self._out_ports.items():
-            if mode in mode_range:
-                return port
-        return None
+        return self.experiment.get_output_port(mode)
 
-    @deprecated(version=0.12, reason="Add detectors as components")
+    # TODO: remove this deprecated method
     def thresholded_output(self, value: bool):
         r"""
         Simulate threshold detectors on output states. All detections of more than one photon on any given mode is
         changed to 1.
 
         :param value: enables threshold detection when True, otherwise disables it.
         """
-        self._thresholded_output = value
-        self._detectors = [Detector.threshold() if self._thresholded_output else None] * len(self._detectors)
+        self.experiment.thresholded_output(value)
 
+    # TODO: remove this deprecated method
     @property
-    @deprecated(version=0.12, reason="Use `detection_type` property instead")
     def is_threshold(self) -> bool:
-        return self._thresholded_output
+        return self.experiment.is_threshold
 
     @property
     def detection_type(self) -> DetectionType:
-        return get_detection_type(self._detectors)
+        return self.experiment.detection_type
 
     @property
     def heralds(self):
-        pos = {}
-        for port, port_range in self._out_ports.items():
-            if isinstance(port, Herald):
-                pos[port_range[0]] = port.expected
-        return pos
-
-    def _with_logical_input(self, input_state: LogicalState):
-        input_state = get_basic_state_from_ports(list(self._in_ports.keys()), input_state)
-        if self._min_detected_photons_filter is None:
-            self._min_detected_photons_filter = input_state.n
-        self.with_input(input_state)
+        return self.experiment.heralds
 
     def check_input(self, input_state: BasicState):
         r"""Check if a basic state input matches with the current processor configuration"""
-        assert self.m is not None, "A circuit has to be set before the input state"
-        expected_input_length = self.m
-        assert len(input_state) == expected_input_length, \
-            f"Input length not compatible with circuit (expects {expected_input_length}, got {len(input_state)})"
-        if input_state.has_polarization:
-            get_logger().warn("Given input state has polarization, that will be ignored in the computation"
-                              " (use with_polarized_input instead).")
-        elif input_state.has_annotations:
-            get_logger().warn("Given input state has annotations, that will be ignored in the computation."
-                              " To use them, consider using a StateVector.")
-
+        self.experiment.check_input(input_state)
```

Comment:
Remove this method from AProcessor ?

Reply:
For this one, the RemoteProcessor does different things than the Processor, so I would say that we don't remove it